packages/compartment-mapper/test/test-cjs-compat.js

```diff
@@ -33,11 +32,25 @@ test('CommonJS exports include a field named `default`', async t => {
     import.meta.url,
   ).toString();
 
-  const application = await loadLocation(read, fixture);
   const {
     namespace: { assertions },
-  } = await application.import({});
+  } = await importLocation(read, fixture);
   assertions.packageWithDefaultField();
   assertions.moduleWithDefaultField();
   t.pass();
 });
+
+test('CommonJS parser detects nested requires', async t => {
+  t.plan(1);
+
+  const fixture = new URL(
+    'fixtures-cjs-compat/node_modules/app/index.js',
+    import.meta.url,
+  ).toString();
+
+  const {
+    namespace: { assertions },
+  } = await importLocation(read, fixture);
+  assertions.parserStruggles();
+  t.pass();
+});
```

Comment:
Please frame this test in terms of `scaffold.js` as seen in `test-main.js` so we can verify that the same behaviors survive a round-trip through an archive.

Reply:
It messes with import-archive. the staticModuleRecord stub returned by postpone is not enough to satisfy its import hook.

Reply:
I fixed bits missing in parse-pre-cjs to get archive working on basic tests, but I don't see how I could get the postponeError functionality to work for archive creation.  
Might have to build a better mental model for the archive build+import flow. Or postpone as far as archive runtime.

Reply:
For postpone to work in an archive, we would need a module record that can be serialized to one of the valid formats and produce the same effect. It does not need to be stack-preserving.

Reply:
My current draft implementation that successfully postpones errors defines this:
```
 * @typedef {Object} ModuleSource
 * @property {string} [deferredError] - module loading error deferred to later stage
 ...
 `

Reply:
done